src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,189 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { TodosContext } from '../../TodosContext';
+import * as todoService from '../../api/todos';
+import { Filter } from '../../types/Filter';
+import { UserWarning } from '../../UserWarning';
+import { TodosList } from '../TodosList';
+import { TodosFilter } from '../TododsFilter';
+import { TempTodoContext } from '../../TempTodoContext';
+
+const USER_ID = 11903;
+const ERROR_DELAY = 3000;
+
+export const TodoApp: React.FC = () => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const { tempTodo, setTempTodo } = useContext(TempTodoContext);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [status, setStatus] = useState('all');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [isHidden, setIsHidden] = useState(true);
+  const refInput = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    todoService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setIsHidden(false);
+        setErrorMessage('Unable to load todos');
+      })
+      .finally(() => {
+        setTimeout(() => setIsHidden(true), ERROR_DELAY);
+      });
+  }, []);
+
+  const filteredTodos = todos.filter(todo => {
+    switch (status) {
+      case Filter.Active:
+        return !todo.completed;
+
+      case Filter.Completed:
+        return todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+
+  const completedTodo = todos.some(todo => todo.completed);
+
+  const handleOnBlur = () => {
+    if (!todoTitle.trim()) {
+      setErrorMessage('Title should not be empty');
+      setIsHidden(false);
+      setTempTodo(null);
+
+      setTimeout(() => {
+        setIsHidden(true);
+      }, ERROR_DELAY);
+    } else {
+      todoService.createTodo({
+        userId: USER_ID,
+        title: todoTitle.trim(),
+        completed: false,
+      })
+        .then(newTodo => {
+          const currentTodos = [...todos];
```

Comment:
no need to copy it here
also, it's better to update the state based on the current in this way: https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state

FYI